apps/frontend/src/components/ui/add-key.tsx

```diff
@@ -209,6 +209,13 @@ export default function AddNewKey({ onClose }: AddNewKeyProps) {
             streamEntryId: streamEntryId.trim() || undefined,
           }),
         )
+      } else if (keyType === "JSON") {
```

Comment:
Let's add consts for all the key types

Reply:
I would change the if/else chain into a switch case statement. I would also only set the payload in each case and dispatch once at the end.